Experimental support for Confluent Cloud

fmck3516 opened the pull request:
Confluent Platform uses serializers/deserializers that add/expect a custom 5-byte prefix before the Avro payload: https://docs.confluent.io/platform/current/schema-registry/serdes-develop/index.html#wire-format

This is currently not supported by your extension. 

I have drafted a PR that shows how to fix that:

- add 5-byte prefix in the producer
- remove 5-byte prefix in the consumer

I've also added a test case:
- spin up a local instance of Confluent Platform
- create the topic `com.example.person`
- register the key and value schema
- adjust the schema id in `producer#Produce` according to the IDs that have been assigned to the registered schemas
- run `./k6 run --vus 1 --duration 1s test/test.js`

The PR can't be merged as is (e.g., I have hard-coded the schema IDs). I would like to start a conversation whether you want to add support for Confluent Platform. I'm happy to contribute a "polished PR" if you are open to the idea.

mostafa commented:
Hi @fmck3516,

Thanks for raising the issue and I'd be happy to review your polished PR.

Also have a look at #3 (#2), where I optimized and fixed some bugs and added SASL Auth. Correct me if I'm wrong, but I think we can make this optional, as I know the current Avro format works as is, but Confluent added their own wire format for schema registry. What do you think?

fmck3516 commented:
> Thanks for raising the issue and I'd be happy to review your polished PR.
Great. I will create the PR some time this week.

> Also have a look at #3 (#2), where I optimized and fixed some bugs and added SASL Auth. 
Will do (SASL support is relevant for us as well).

> Correct me if I'm wrong, but I think we can make this optional, as I know the current Avro format works as is, but Confluent added their own wire format for schema registry. What do you think?

That's correct: Support for the wire format with the proprietary 5-byte prefix has to be optional.
